pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -188,6 +183,25 @@ func (c *Cluster) Start() error {
 			}
 		}
 	}
+	deviceSetNames := map[string]bool{}
+	for _, deviceSet := range c.spec.Storage.StorageClassDeviceSets {
+		if deviceSetNames[deviceSet.Name] {
```

Comment:
if the osds are created by set-1 and then set-2 is added with same device-set, it will block the reconcile of the first device-set too,
Instead can we not have the x-validator on the CRD?

Reply:
Not able to get the rule, spend sometime 
```
	// +kubebuilder:validation:XValidation:message="do not duplicate the deviceset name",rule="!has_duplicates(self.StorageClassDeviceSets[*].name)"
	StorageClassDeviceSets []StorageClassDeviceSet `json:"storageClassDeviceSets,omitempty"`
```
cc @subhamkrai if you have any idea

Reply:
Btw if go by first approach I want to make sure we do not stop the reconcile for older created osds

Reply:
The `has_duplicates()` function is not available, I get a syntax error with that suggested rule, and I can't see how to get the validation rule working in this case. On the [validation doc](https://kubernetes.io/blog/2022/09/23/crd-validation-rules-beta/) there is an example of setting the type `x-kubernetes-list-type` and then use a validation rule to confirm two sets are disjoint, but not sure how to apply this to check for unique names in the same set.

Reply:
> Btw if go by first approach I want to make sure we do not stop the reconcile for older created osds

Actually I am intentionally avoiding the reconcile of existing OSDs as well. If there is a duplicate set name we don't really know which of the duplicates is valid, so it's better to skip all reconcile of OSDs and force the user to fix it before continuing.